### DISCUSSION TITLE
SAHI with line (LETR)

### DISCUSSION THREAD

**Discussion by aquexel** [obss/sahi#742] [Q&A]
Hello,

I'm using LETR for line detection with large pictures. Is it possible to use SAHI to split dataset (pictures too) with lines (labeled on Labelme) and do inference on large pictures ?

Thanks for all !

**Comment by fcakyon** [comment]
What is your label format? SAHI currently supports coco object detection and instance segmentation annotations.

**Reply by aquexel** [reply]
Hi
Actually format label is json like this:
{
  "version": "4.0.0",
  "flags": {},
  "shapes": [
    {
      "label": "bus",
      "points": [
        [
          260.936170212766,
          23.33306055646483
        ],
        [
          193.936170212766,
          20.33306055646483
        ],
        [
          124.93617021276599,
          40.33306055646483
        ],
        [
          89.93617021276599,
          102.33306055646483
        ],
        [
          81.93617021276599,
          151.33306055646483
        ],
        [
          108.93617021276599,
          146.33306055646483
        ],
        [
          88.93617021276599,
          245.33306055646483
        ],
        [
          89.93617021276599,
          323.33306055646483
        ],
        [
          116.93617021276599,
          368.33306055646483
        ],
        [
          158.936170212766,
          369.33306055646483
        ],
        [
          165.936170212766,
          338.33306055646483
        ],
        [
          347.936170212766,
          336.33306055646483
        ],
        [
          349.936170212766,
          370.33306055646483
        ],
        [
          391.936170212766,
          374.33306055646483
        ],
        [
          403.936170212766,
          336.33306055646483
        ],
        [
          425.936170212766,
          333.33306055646483
        ],
        [
          421.936170212766,
          282.33306055646483
        ],
        [
          428.936170212766,
          253.33306055646483
        ],
        [
          428.936170212766,
          237.33306055646483
        ],
        [
          409.936170212766,
          221.33306055646483
        ],
        [
          409.936170212766,
          151.33306055646483
        ],
        [
          430.936170212766,
          144.33306055646483
        ],
        [
          433.936170212766,
          113.33306055646483
        ],
        [
          431.936170212766,
          97.33306055646483
        ],
        [
          408.936170212766,
          91.33306055646483
        ],
        [
          395.936170212766,
          51.33306055646483
        ],
        [
          338.936170212766,
          26.33306055646483
        ]
       ],
      "group_id": null,
      "shape_type": "lines",
      "flags": {}
    }
  ],
  "imagePath": "2011_000025.jpg",
  "imageData": null,
  "imageHeight": 375,
  "imageWidth": 500
}

**Reply by fcakyon** [reply]
That format is not supported, you can use the utilities from sahi to implement your custom scripts.

**Reply by aquexel** [reply]
Okay thank you,
If I convert lines to polygon with two points in COCO format will it work for inference and split ?
I don't know how the box will it work with lines 
thanks

**Reply by fcakyon** [reply]
Check this url for object detection format: https://cocodataset.org/#format-data

You can represent lines as polygons (at least 3 points required), and set the `segmentation` field with point coords.

**Reply by aquexel** [reply]
Thanks,
I check that
